### DISCUSSION TITLE
Is there an example of using a model with an OpenAI style? (e.g. LocalAI)?

### DISCUSSION THREAD

**Discussion by shanecowherd** [intitni/CopilotForXcode#355] [General]
Does this mean we can use a local app (that matches ChatGPT api) to allow usage without connecting to the internet?

Is there an example of setting this up with something like LlamaCode or Faraday or something?

Awesome!

**Comment by intitni** [comment]
Yes.

To use LocalAI, you can 
1. Clone the repo [repo](https://github.com/go-skynet/LocalAI).
2. Download a model and place it into the models folder. [EASY MODEL IMPORT - DOWNLOADED](https://localai.io/howtos/easy-model-import-downloaded/)
3. Run LocalAI with docker, or [build and run it locally](https://localai.io/basics/build/).
4. Setup the model
<img width="671" alt="Screenshot 2023-09-07 at 14 05 15" src="https://github.com/intitni/CopilotForXcode/assets/793147/255c6eeb-e4de-4103-83dd-462c80c27f19">

I have only tried building and running locally. 

But honestly speaking, it's super slow.
